drivers: tee: optee: add include for kernel_arch_interface

oleksiimoisieiev opened the pull request:
Build fails with the following error:
> fatal error: kernel_arch_interface.h: No such file or directory

It was found that required header is placed in
${ZEPHYR_BASE}/kernel/include directory so add it to the CMakeLists file.

oleksiimoisieiev committed:
drivers: tee: optee: add include for kernel_arch_interface

Build fails with the following error:
> fatal error: kernel_arch_interface.h: No such file or directory

It was found that required header is placed in
${ZEPHYR_BASE}/kernel/include directory so add it to the CMakeLists file.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

lorc reviewed (approved):
`Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

firscity reviewed (approved):
`Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>`

oleksiimoisieiev committed:
drivers: tee: optee: handle configuration with MMU disabled.

For configurations without MMU support, such as native_posix_64
which is used for the unit test. So use z_mem_phys_addr for non-mmu
configurations.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
